reward/src/lib.rs

```diff
@@ -0,0 +1,536 @@
+                    amount = U256::from_dec_str(&amt).ok();
+                }
+                Long("strategy") => {
+                    strategy = parser.value()?.parse().ok();
+                }
+                _ => return Err(anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((
+            Options {
+                range,
+                dry,
+                amount,
+                token,
+                strategy,
+                provider,
+                signer,
+            },
+            vec![],
+        ))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let rt = tokio::runtime::Runtime::new()?;
+    let provider = ethereum::provider(options.provider)?;
+    let signer_opts = options.signer;
+    let (wallet, provider) = rt.block_on(term::ethereum::get_wallet(signer_opts, provider))?;
+    let signer: Arc<_> = SignerMiddleware::new(provider, wallet).into();
+    let profile = profile::default()?;
+    let storage = profile::read_only(&profile)?;
+    let repo = git::Repository::open(Path::new("."))?;
+
+    let mut revwalk = repo.revwalk()?;
+    let range = options
+        .range
+        .or_else(|| {
+            if !repo.tag_names(Some("rad/reward")).ok()?.is_empty() {
+                return Some("rad/reward..".to_string());
+            }
+            None
+        })
+        .ok_or_else(|| anyhow!("Can't determine the range of commits"))?;
+    revwalk.push_range(&range)?;
+
+    // from identities in local monorepo build a map of: fp -> ens
+    let mut ssh_to_ens: HashMap<String, _> = HashMap::new();
+    for identity in identities::any::list(&storage)?.flatten() {
+        if let Person(person) = identity {
+            let peer_ids = person
+                .delegations()
+                .iter()
+                .map(|key| -> PeerId { key.to_owned().into() })
+                .collect::<Vec<PeerId>>();
+            if let Ok(Some(ens)) = person.payload().get_ext::<Ens>() {
+                for peer_id in peer_ids {
+                    let fp = keys::to_ssh_fingerprint(&peer_id)?;
+                    ssh_to_ens.insert(fp.to_string(), ens.clone());
+                }
+            }
+        }
+    }
+
+    term::headline("Retrieving commits...");
+
+    // collect data regarding provided range
+    let range_data = revwalk
+        .filter_map(|sha1| {
+            let sha1 = match sha1 {
+                Err(_) => return None,
+                Ok(sha1) => sha1,
+            };
+            let fp = git::commit_ssh_fingerprint(Path::new("."), &sha1.to_string()).ok()?;
+            let ens = match fp {
+                Some(ref fp) => ssh_to_ens.get(fp),
+                None => None,
+            };
+            Some((sha1, fp, ens))
+        })
+        .collect::<Vec<_>>();
+
+    let all_ens = range_data
+        .iter()
+        .filter_map(|(_, _, ens)| *ens)
+        .collect::<Vec<_>>();
+
+    let head_sha1 = range_data.get(0).map(|(sha1, _, _)| *sha1);
```

Comment:
FYI you can also use `first()` instead of `get(0)`

Reply:
https://github.com/radicle-dev/radicle-cli/compare/30805149493b3e5bd4eebcd5afb5746c9b987c3f..e9e132c44330b15ae72aa3ea6dbbae51b7d2058f#diff-26d7e6053383a7793e880ae8b6ee5d2f7d7c23ccd5e88e43f44052374e9f0f58R209